### PR TITLE
[Parser] Enable dynamic loading in gRPC server

### PR DESCRIPTION
**Pull request by tonyyang-svail**
This PR implements the API design https://github.com/sql-machine-learning/sqlflow/issues/1904#issuecomment-593727993.

### CODE REVIEW

File: java/parser/src/main/java/org/sqlflow/parser/ParserFactory.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        className = className.replace('/', '.');
+
+        if (className.startsWith("org.sqlflow")
+            && !className.startsWith("org.sqlflow.parser.parse")) {
+          Class c = cl.loadClass(className);
+          for (Class<?> x : c.getInterfaces()) {
+            if ("org.sqlflow.parser.parse.ParseInterface".equals(x.getName())) {
+              Object inst = c.getConstructor().newInstance();
+              ParseInterface parser = (ParseInterface) inst;
+              parsers.put(parser.dialect(), c);
```
Comment (typhoonzero) [inline java/parser/src/main/java/org/sqlflow/parser/ParserFactory.java:52]: Maybe can log the loaded classes?

**Review by typhoonzero — DISMISSED** [review]
Cool, LGTM

**Review by Yancey0623 — DISMISSED** [review]

File: java/parser/src/main/java/org/sqlflow/parser/ParserFactory.java (hunk 1 above)
Comment (tonyyang-svail) [reply java/parser/src/main/java/org/sqlflow/parser/ParserFactory.java:52]: Sounds good.

**Review by weiguoz — APPROVED** [review]
LGTM

**Pull request merged**